source/grid/grid_in.cc

```diff
@@ -3060,7 +3060,10 @@ GridIn<dim, spacedim>::parse_tecplot_header(
   blocked    = false;
 
   // convert the string to upper case
-  std::transform(header.begin(), header.end(), header.begin(), ::toupper);
+  std::transform(header.begin(),
+                 header.end(),
+                 header.begin(),
+                 static_cast<int (*)(int)>(std::toupper));
```

Comment:
Here and elsewhere in this file: `::toupper` is simply a function. For `std::toupper` that's not clear to me. https://en.cppreference.com/w/cpp/string/byte/toupper says that it's also just a single function, but when passed here in place of a template argument to `std::transform`, the compiler complains about it being overloaded. ... oh, on further search, there is a second overload that comes from here: https://en.cppreference.com/w/cpp/locale/toupper ... In any case, the cast ensures that we pick exactly one of the overload set.

Reply:
Also discussed in https://stackoverflow.com/questions/7131858/stdtransform-and-toupper-no-matching-function.